src/Cli/dotnet/CommonOptions.cs

```diff
@@ -272,6 +272,42 @@ private static IEnumerable<string> ForwardSelfContainedOptions(bool isSelfContai
             parseResult.HasOption(ArchitectureOption) ||
             parseResult.HasOption(LongFormArchitectureOption) ||
             parseResult.HasOption(OperatingSystemOption);
+
+        internal static string BuildProperty<TValue>(string name, TValue value, bool escapeValue = false)
+        {
+            var valueString = value.ToString();
+
+            var needsToBeEscaped = escapeValue
+                && !valueString.StartsWith("\"", StringComparison.Ordinal)
+                && !valueString.EndsWith("\"", StringComparison.Ordinal);
+
+            return needsToBeEscaped
+                ? $"-property:{name}=\"{valueString}\""
```

Comment:
minor nit - for user-supplied properties we've standardized on forwarding those to MSBuild as `--property:<VALUE>`, could this helper do the same? just to make sure it all looks uniform.